nylas/client/restful_models.py

```diff
@@ -1,17 +1,38 @@
-from datetime import datetime
+from datetime import datetime, date
 
 from nylas.client.restful_model_collection import RestfulModelCollection
 from nylas.client.errors import FileUploadError
 from nylas.utils import timestamp_from_dt
 from six import StringIO
 
+try:
+    from werkzeug.datastructures import MultiDict
+except ImportError:  # pragma: no cover
+    MultiDict = dict
+
 # pylint: disable=attribute-defined-outside-init
 
+def detect_attr(value):
+    attr_type = value.get("object", None)
+    if attr_type == "date":
+        date_str = value["date"]
+        if date_str:
+            return datetime.strptime(date_str, "%Y-%m-%d").date()
+    return None
+
+
+def mdict_attr(items, attr_name=None):
```

Comment:
My takeaways from our discussion yesterday:
- If we use MultiDict, we should make the dependency mandatory. The fallback to dict would only keep the last value of each type, but 95% of our users should care about the full list for each type. 
- The only benefit of MultiDict over parsing our own dict is that MultiDict allows users to get a single value for a type.

I agree that the typed dict is useful, but I don't think the 'single value' functionality is useful enough to warrant introducing the MultiDict dependency/different syntax. I think we should parse the data into a normal dict, and rename `mdict_attr` to something like `typed_dict_attr`.